src/Illuminate/Database/MySqlConnection.php

```diff
@@ -76,7 +76,8 @@ public function bindValues($statement, $bindings)
     {
         foreach ($bindings as $key => $value) {
             $statement->bindValue(
-                is_string($key) ? $key : $key + 1, $value,
+                is_string($key) ? $key : $key + 1,
+                is_array($value) || is_object($value) ? json_encode($value) : $value,
```

Comment:
Please move the JSON encoding to `Grammar::prepareBindingsForUpdate()`. We don't need to check every binding of every query.

We then have to exclude `Expression` objects from being encoded (using `$this->isExpression()`).

Reply:
I moved it to the `MySqlGrammar` class and not the parent `Grammar` because `PosgresGrammar` already has its own logic for that.

Reply:
`PostgresGrammar` doesn't handle cases without a `->` operator: `->update(['meta' => ['options' => ['foo' => 'bar']]])`

My idea was to cover both databases in `Grammar`, but I just noticed that this doesn't work.

Could also add the PostgreSQL implementation?

Reply:
I added the implementation to `PostgresGrammar` and the tests are OK. However please note two things:

- In the `prepareBindingsForUpdate` test, I had to remove the extra quotations. I think this should be still good, but at the moment it can't test it with a real database. Does it really need the extra `"`-s? Please, confirm if this works or if I need to revert it.
- Now, in both `MySqlGrammar` and `PostgresGrammar` the $column variable is useless. We may remove it. What do you think?

Thanks!

Reply:
The quotes are absolutely necessary, the original test still has to work. You can't remove `$this->isJsonSelector($column)`.

You can remove `$column` from `MySqlGrammar`.

Reply:
Now I'm a bit confused about Postgres. What do we need to implement exactly? 
Because it encodes the object and array values to JSON by default.

Or you meant something else?

Reply:
`PostgresGrammar` doesn't handle cases without a `->` operator: `->update(['meta' => ['options' => ['foo' => 'bar']]])`

What I mean:

```php
! $this->isExpression($value) && ($this->isJsonSelector($column) || is_array($value) || is_object($value))
```

Reply:
Ok, got you, thanks! It was not clear for me, sorry.
Now the original tests are passing!

Reply:
👍 Please also add a test for PostgreSQL.

Reply:
Yes, I'm on it! Soon commit.

Reply:
I added the test for PostgreSQL.